File: pywebsocket/MyService/TestService.py
from abc import ABC, abstractmethod
from MyService.TestMessage import TestMessage as ImportedRef__MyService_TestMessage
from UniRpc.BaseMessage import BaseMessage as ImportedRef__UniRpc_BaseMessage
from UniRpc.WebsocketServiceBase import WebsocketServiceBase as ImportedRef__UniRpc_WebsocketServiceBase

class TestService(ImportedRef__UniRpc_WebsocketServiceBase):
    def __init__(self):
        super().__init__()
        self.__reflection = 'MyService.TestService'

    @abstractmethod
    async def test(self, name: str, value: float, msg: ImportedRef__MyService_TestMessage) -> None:
        pass

    async def __invoke(self, message: ImportedRef__UniRpc_BaseMessage) -> ImportedRef__UniRpc_BaseMessage:
        if message.Method == 'test':
            ____name: str = message.Payload['name']
            ____value: float = message.Payload['value']
            ____msg: ImportedRef__MyService_TestMessage = message.Payload['msg']
            await self.test(____name, ____value, ____msg)
        else:
            raise Exception(f'{message.Service}.{message.Method} is not defined.')

File: jswebsocket/MyService/TestService.ts
import {BaseMessage as $UniRpc_BaseMessage} from "../UniRpc/BaseMessage";
import {WebsocketServiceBase as $UniRpc_WebsocketServiceBase} from "../UniRpc/WebsocketServiceBase";
import {TestMessage as $MyService_TestMessage} from "./TestMessage";
export abstract class TestService extends $UniRpc_WebsocketServiceBase
{
    public constructor () {
        super();
        this.__reflection = "MyService.TestService";
    }
    public abstract test(name: string, value: number, msg: $MyService_TestMessage): Promise<void>;
    public async __invoke(message: $UniRpc_BaseMessage): Promise<$UniRpc_BaseMessage> {
        switch (message.Method)
        {
            case "test":
                {
                    let ____name: string = message.Payload['name'];
                    let ____value: number = message.Payload['value'];
                    let ____msg: $MyService_TestMessage = message.Payload['msg'];
                    await this.test(____name, ____value, ____msg);
                    break;
                }
        }
        throw `${message.Service}.${message.Method} is not defined.`;
    }
}